### PR TITLE
Logger configuration

### PR DESCRIPTION
**Pull request by PavelLoparev**

### CODE REVIEW

File: inc/Smartling/MonologWrapper/Logger/LevelLogger.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   *
+   * @var int level
+   */
+  private $level = Logger::DEBUG;
+
+  /**
+   * @var array
+   */
+  private $levelMapping = [
+    LogLevel::EMERGENCY => Logger::EMERGENCY,
```
Comment (dimitrystd) [inline inc/Smartling/MonologWrapper/Logger/LevelLogger.php]: Why do we need to introduce own log level?

Comment (PavelLoparev) [reply inc/Smartling/MonologWrapper/Logger/LevelLogger.php]: This is not custom level. It's one of default 8 levels in monolog. Monolog is RFC5424 compliant so there are debug, info, notice, warning, error, critical, alert and emergency levels.

File: inc/Smartling/Helpers/SiteHelper.php [hunk 2]
```diff
@@ -35,11 +36,11 @@ public function getInitialBlogId()
...
 
     /**
-     * @param LoggerInterface $logger
+     * Instantiates SiteHelper object.
      */
-    public function __construct(LoggerInterface $logger)
+    public function __construct()
     {
-        $this->_logger = $logger;
+        $this->_logger = MonologWrapper::getLogger(get_called_class());;
```
Comment (dimitrystd) [inline inc/Smartling/Helpers/SiteHelper.php]: Let's rename field `_logger` to `logger` for consistency. Also add and use `getLogger`.

File: inc/Smartling/Helpers/XmlEncoder.php [hunk 3]
```diff
@@ -30,15 +30,12 @@ private static function getFieldProcessingParams()
...
-     * Controlled by logger.smartling_verbose_output_for_xml_coding bool value
      *
      * @param $message
      */
     public static function logMessage($message)
     {
-        if (true === (bool)Bootstrap::getContainer()->getParameter('logger.smartling_verbose_output_for_xml_coding')) {
-            Bootstrap::getLogger()->debug($message);
-        }
+        Bootstrap::getLogger()->debug($message);
```
Comment (dimitrystd) [inline inc/Smartling/Helpers/XmlEncoder.php]: Don't see a reason to use logger from bootstrap now. We can create logger for this class.

File: inc/Smartling/MonologWrapper/Logger/LevelLogger.php (hunk 1 above)
Comment (dimitrystd) [reply inc/Smartling/MonologWrapper/Logger/LevelLogger.php]: Didn't understand but ok :)

Comment (PavelLoparev) [reply inc/Smartling/MonologWrapper/Logger/LevelLogger.php]: Oh, you might be talking about `$levelMapping` but not `EMERGENCY` string. Yes, `$levelMapping`  is redundant and will be removed.

Comment (PavelLoparev) [reply inc/Smartling/MonologWrapper/Logger/LevelLogger.php]: Fixed

File: inc/Smartling/DbAl/Migrations/DbMigrationManager.php [hunk 4]
```diff
@@ -12,6 +13,15 @@
 class DbMigrationManager extends SmartlingFactoryAbstract
 {
 
+    /**
+     * DbMigrationManager constructor.
+     */
+    public function __construct()
+    {
+        $logger = MonologWrapper::getLogger(get_called_class());
```
[the remark this answers is not in the document]
Comment (PavelLoparev) [reply inc/Smartling/DbAl/Migrations/DbMigrationManager.php]: You're right. Fixed in all derived from `SmartlingFactoryAbstract` classes.

File: inc/Smartling/DbAl/MultiligualPressConnector.php [hunk 5]
```diff
@@ -129,12 +130,13 @@ public function getBlogLocaleById($blogId)
...
      * @inheritdoc
      */
-    public function __construct(LoggerInterface $logger, SiteHelper $helper, array $ml_plugin_statuses)
+    public function __construct(SiteHelper $helper, array $ml_plugin_statuses)
     {
         global $wpdb;
 
         $this->wpdb = $wpdb;
 
+        $logger = MonologWrapper::getLogger(get_called_class());
```
[the remark this answers is not in the document]
Comment (PavelLoparev) [reply inc/Smartling/DbAl/MultiligualPressConnector.php]: Fixed

File: inc/Smartling/Helpers/SiteHelper.php (hunk 2 above)
Comment (PavelLoparev) [reply inc/Smartling/Helpers/SiteHelper.php]: Fixed

File: inc/Smartling/Helpers/XmlEncoder.php (hunk 3 above)
Comment (PavelLoparev) [reply inc/Smartling/Helpers/XmlEncoder.php]: Fixed

**Pull request merged**